### PR TITLE
Feature/AC-415

### PR DESCRIPTION
**Pull request by de4db33f**
- Interceptor for api key header to retrofit

### CODE REVIEW

File: app/src/main/java/com/aplication/carsales/main_module/model/RemoteApi.kt [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
+        .client(client)
         .addConverterFactory(GsonConverterFactory.create())
         .build()
 
     private val service = retrofit.create(CovidService::class.java)
 
-    suspend fun getCovidDataFromDate(date: String) : CovidDataEntity = withContext(Dispatchers.IO){
+    suspend fun getCovidDataFromDate(date: String): CovidDataEntity = withContext(Dispatchers.IO) {
         service.getCovidDataFromDate(date)
     }
```
Comment (aapavezcs) [inline app/src/main/java/com/aplication/carsales/main_module/model/RemoteApi.kt]: Could you implement Singleton Pattern in RemoteApi? 
if you have only one instance (Retrofit), It is better . 👍

File: app/src/main/java/com/aplication/carsales/main_module/view_model/MainViewModel.kt [hunk 2]
```diff
@@ -21,11 +22,11 @@ class MainViewModel: ViewModel() {
     private val loading = MutableLiveData<Boolean>()
     fun isLoaded() = loading
 
-    suspend fun getCovidDataFromDate(date: String){
+    suspend fun getCovidDataFromDate(date: String) {
         viewModelScope.launch {
             try {
                 loading.value = false
```
Comment (aapavezcs) [inline app/src/main/java/com/aplication/carsales/main_module/view_model/MainViewModel.kt]: If you use viewModelScope.launch, you don't need to implement "suspend function"

**Review by aapavezcs — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/aplication/carsales/main_module/model/RemoteApi.kt (hunk 1 above)
Comment (de4db33f) [reply app/src/main/java/com/aplication/carsales/main_module/model/RemoteApi.kt]: It's OK now!

File: app/src/main/java/com/aplication/carsales/main_module/view_model/MainViewModel.kt (hunk 2 above)
Comment (de4db33f) [reply app/src/main/java/com/aplication/carsales/main_module/view_model/MainViewModel.kt]: True. Fix it!

**Review by aapavezcs — APPROVED** [review]

**Pull request merged**